digdag-core/src/test/java/io/digdag/core/database/DatabaseSecretStoreTest.java

```diff
@@ -82,6 +82,18 @@ public void singleSecret()
     }
 
     @Test
+    public void lockSecret()
+            throws Exception
+    {
+        Optional<String> absent = factory.begin(() -> secretControlStore.lockProjectSecret(projectId, SecretScopes.PROJECT, KEY1, (control, value) -> value));
+        assertThat(absent, is(Optional.absent()));
+
+        factory.begin(() -> secretControlStore.setProjectSecret(projectId, SecretScopes.PROJECT, KEY1, VALUE1));
+        Optional<String> present = factory.begin(() -> secretControlStore.lockProjectSecret(projectId, SecretScopes.PROJECT, KEY1, (control, value) -> value));
```

Comment:
We'd better have a test that confirms if another thread is blocked with the method